main.go

```diff
@@ -357,3 +379,26 @@ func envOrDefault(envName, defaultValue string) string {
 
 	return defaultValue
 }
+
+func registryClientGenerator(isLogin bool) (*registry.Client, string, error) {
```

Comment:
This does not appear to be actually used.

Reply:
We could move this to somewhere else (some internal pkg?), as this would need to be reused here and within the tests.

Reply:
Yes, I would feel more comfortable about that. `main.go` is already looking pretty brutal.

As a separate issue: feels like some of the other private functions could also be factored out. `mustInit` could simply be moved to `controllers/storage.go`. (For now, factoring the storage out and making it also work as a client utility with caching would be even better). In addition, `startFileServer` feels a bit too lean to be comfortable with.